reference/debugging.md

```diff
@@ -0,0 +1,284 @@
+
+    The terminal will print a typical file list from a pre-0.50 Ambassador instance, similar to the following:
+
+    ```console
+    $ kubectl get pods
+    NAME                         READY     STATUS    RESTARTS   AGE
+    ambassador-85c4cf67b-4pfj2   1/1       Running   0          14m
+    ambassador-85c4cf67b-fqp9g   1/1       Running   0          14m
+    ambassador-85c4cf67b-vg6p5   1/1       Running   0          14m
+    $
+    $ kubectl exec -it ambassador-85c4cf67b-4pfj2 -- /bin/sh
+    /ambassador # pwd
+    /ambassador
+    /ambassador # ls -lsa
+    total 84
+        4 drwxrwxr-x    1 root     root          4096 Oct 15 12:35 .
+        4 drwxr-xr-x    1 root     root          4096 Oct 15 12:26 ..
+        4 drwxr-xr-x    4 root     root          4096 Oct 15 12:26 ambassador-0.40.0-py3.6.egg-tmp
+        4 drwxrwxr-x    1 root     root          4096 Sep 25 20:29 ambassador-config
+        4 drwxr-xr-x    2 root     root          4096 Oct 15 12:26 ambassador-config-1
+        4 drwxr-xr-x    2 root     root          4096 Oct 15 12:35 ambassador-config-2
+        4 drwxrwxr-x    1 root     root          4096 Sep 25 20:29 ambassador-demo-config
+        8 -rwxr-xr-x    1 root     root          4179 Sep 25 20:28 entrypoint.sh
+        4 -rw-r--r--    1 root     root          3322 Oct 15 12:26 envoy-1.json
+        8 -rw-r--r--    1 root     root          4101 Oct 15 12:35 envoy-2.json
+        8 -rw-rw-r--    1 root     root          5245 Sep 25 20:28 hot-restarter.py
+    20 -rw-rw-r--    1 root     root         16952 Sep 25 20:28 kubewatch.py
+        4 -rwxrwxr--    1 root     root           175 Sep 25 20:28 requirements.txt
+        4 -rwxr-xr-x    1 root     root           997 Sep 25 20:28 start-envoy.sh
+    ```
+
+The `ambassador-config-N` lines indicate the directories which contain the specific Ambassador configuration used in YAML files during updates. Higher numbers indicate the most recent configuration, as verified by the directory timestamp.
+
+2. Determine the latest configuration. An easy method is to simply look for the directory with the highest number. For example, `ambassador-config-2` is the most recent Ambassador configuration.
+3. Print the contents of the latest configuration: `/ambassador # ls ambassador-config-2`
+4. Navigate into the YAML file within that directory: `/ambassador # cat ambassador-config-2/Httpbin-default.yaml`
+
+    The terminal will print something similar to the following:
+
+    ```yaml
+    ---
+    apiVersion: v0.1
+    kind: Pragma
+    ambassador_id: default
+    source: "service httpbin, namespace default"
+    autogenerated: true
+    ---
+    apiVersion: ambassador/v1
+    kind:  Mapping
+    name:  httpbin_mapping
+    prefix: /httpbin/
+    service: httpbin.org:80
+    host_rewrite: httpbin.org
+    ```
+
+5. Fix any errors you find in this YAML file.
```

Comment:
This would be more like "if you see anything wrong with the `snapshot` or the `aconf`, you need to fix your configuration. If you see something wrong with the `ir` or `econf`, you need to open an Ambassador issue."